core/server/data/migrations/versions/4.35/2022-02-02-13-10-transform-specific-tiers-default-content-visibility.js

```diff
@@ -0,0 +1,114 @@
+        if (!contentVisibilitySetting || !contentVisibilitySetting.value) {
+            logging.warn('No default_content_visibility setting found.');
+            return;
+        }
+
+        if (!visibilityTiersSetting) {
+            logging.warn('No default_content_visibility_tiers setting found.');
+            return;
+        }
+
+        const contentVisibility = contentVisibilitySetting.value;
+        // Transform visibility to tiers when stored as nql string
+        if (!['public', 'members', 'paid'].includes(contentVisibility)) {
+            const isValidProductNqlFilter = /^(?:product:[\w-]+,?)+$/.test(contentVisibility);
+            let newVisibilityValue = 'paid';
+            let newVisibilityTiersValue = [];
+            if (isValidProductNqlFilter) {
+                newVisibilityValue = 'tiers';
+                // fetch product slugs from nql filter
+                const productSlugs = contentVisibility.split(',').map((segment) => {
+                    return segment.replace('product:', '');
+                });
+
+                // get product ids for slugs
+                const products = await knex('products')
+                    .select('id')
+                    .whereIn('slug', productSlugs);
+                newVisibilityTiersValue = products.map((product) => {
+                    return product.id;
+                });
+            }
+
+            logging.info(`Updating default_content_visibility to tiers`);
+            await knex('settings')
+                .where({
+                    key: 'default_content_visibility'
+                })
+                .update({
+                    value: newVisibilityValue
+                });
+
+            logging.info(`Updating default_content_visibility_tiers to ${newVisibilityTiersValue}`);
+            await knex('settings')
+                .where({
+                    key: 'default_content_visibility_tiers'
+                })
+                .update({
+                    value: JSON.stringify(newVisibilityTiersValue)
+                });
+        } else {
+            logging.info(`Ignoring default_content_visibility change as set to ${contentVisibility}.`);
+        }
+    },
+    async function down(knex) {
+        logging.info('Reverting default_content_visibility for specific tiers');
+
+        const settings = await knex('settings')
+            .select()
+            .whereIn('key', ['default_content_visibility', 'default_content_visibility_tiers']);
+        const contentVisibilitySetting = settings.find(d => d.key === 'default_content_visibility');
+        const visibilityTiersSetting = settings.find(d => d.key === 'default_content_visibility_tiers');
+
+        const visibilityValue = contentVisibilitySetting && contentVisibilitySetting.value;
+        const visibilityTiersValue = visibilityTiersSetting && visibilityTiersSetting.value;
+
+        if (visibilityValue === 'tiers' && visibilityTiersValue) {
```

Comment:
Preferably the same here with inverting the if-statement to early return

Reply:
✅